From c31072e3c39c70a17d107b08af53e33debbae432 Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Wed, 27 Jan 2021 19:23:48 +0800
Subject: [PATCH] drm/rockchip: vop2: Change wb RGB888 to BGR888

The actually writeback format of 24bit RGB(without alpha)
is BGR888, the TRM description is wrong.

Change-Id: I7897eba9024efc7f95bda57743ff6c4d8533eeb0
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop2.c | 6 +++---
 drivers/gpu/drm/rockchip/rockchip_vop2_reg.c | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
index 794f6f252c21..785e2a61a0e4 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
@@ -342,7 +342,7 @@ struct vop2_wb {
 
 enum vop2_wb_format {
 	VOP2_WB_ARGB8888,
-	VOP2_WB_RGB888,
+	VOP2_WB_BGR888,
 	VOP2_WB_RGB565,
 	VOP2_WB_YUV420SP = 4,
 	VOP2_WB_INVALID = -1,
@@ -891,8 +891,8 @@ static enum vop2_wb_format vop2_convert_wb_format(uint32_t format)
 	switch (format) {
 	case DRM_FORMAT_ARGB8888:
 		return VOP2_WB_ARGB8888;
-	case DRM_FORMAT_RGB888:
-		return VOP2_WB_RGB888;
+	case DRM_FORMAT_BGR888:
+		return VOP2_WB_BGR888;
 	case DRM_FORMAT_RGB565:
 		return VOP2_WB_RGB565;
 	case DRM_FORMAT_NV12:
diff --git a/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c b/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c
index 706bd9ead607..7ea242af0153 100644
--- a/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c
+++ b/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c
@@ -54,7 +54,7 @@ static const uint32_t formats_win_lite[] = {
 };
 
 static const u32 formats_wb[] = {
-	DRM_FORMAT_RGB888,
+	DRM_FORMAT_BGR888,
 	DRM_FORMAT_ARGB8888,
 	DRM_FORMAT_RGB565,
 	DRM_FORMAT_NV12,
-- 
2.35.3

